pkg/authorize/ssl/ssl.go

```diff
@@ -52,7 +63,57 @@ func ClientCertInfoAsHeaders(config ClientCertConfig, logger log.Logger) func(ht
 				http.Error(w, "invalid secret", http.StatusForbidden)
 				return
 			}
-			next.ServeHTTP(w, r)
+
+			cnInfo := r.Header.Get(config.Config.CommonNameHeader)
+			if cnInfo == "" {
+				level.Info(logger).Log("msg", "invalid format for organisation and CN", "cnInfo", cnInfo)
+				http.Error(w, fmt.Sprintf("subject is empty. organisation and common name must be sent in request header %s",
+					config.Config.CommonNameHeader), http.StatusForbidden)
+				return
+			}
+
+			// cnInfo must be provided in the format of "/O = xyz, /CN = 123
+			parts := strings.Split(cnInfo, ",")
+			if len(parts) != 2 {
+				level.Info(logger).Log("msg", "invalid format for organisation and CN", "cnInfo", cnInfo)
+				http.Error(w, fmt.Sprintf("invalid format for organisation and CN %s",
+					config.Config.CommonNameHeader), http.StatusForbidden)
```

Comment:
Is there a mistake in the error string? I guess you wanted `invalid format for organisation and CN in http header %s`